[chore] Updates docker api to version v25 from v24, testcontainers-go from v0.20.0 to v0.29.1

hughesjj opened the pull request:
**Description:** 

I don't have permissions to directly add commits to the update PR so running it here

- [PR](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/31345)
- [Issue](https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/31087)

atoulme commented:
@hughesjj please try running `make for-all CMD=tidy`

hughesjj committed:
remove replace directives for docker/docker from flink and kafkametrics receivers

hughesjj committed:
Use container.ListOptions over deprecated dtypes.ContainerListOptions

crobert-1 commented:
> `Error: OCI runtime error: chmod `run/shm`: Operation not supported` 😶 No idea what that means for this PR

Probably frequency of https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/31443 which should be resolved, are you fully rebased on main?

atoulme commented:
> Error: OCI runtime error: chmod run/shm: Operation not supported 😶
I fixed this for the deb build with https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/31527

hughesjj committed:
Remove replace directives for docker/docker

hughesjj committed:
Update to testcontainers-go@v0.29.1 and docker/docker@v25.0.5+incompatible
Ran the following
```
find -name go.mod -exec grep -l testcontainers/testcontainers-go {}  \; -execdir go get -x github.com/testcontainers/testcontainers-go@v0.29.1 \; \
&& find -name go.mod -exec grep -l testcontainers/testcontainers-go {}  \; -execdir go mod tidy -compat=1.21 \; \
&& find -name go.mod -exec grep -l docker/docker {} \; -execdir go get -x github.com/docker/docker@v25.0.5+incompatible \; \
&& find -name go.mod -exec grep -l docker/docker {}  \; -execdir go mod tidy -compat=1.21 \;
```
then in internal/metadataproviders ran the following in response to go complaints
```
go get github.com/distribution/reference # Claimed this needed import
go get google.golang.org/genproto/googleapis/api/httpbody # claimed ambiguous
go get google.golang.org/genproto/googleapis/rpc/status # claimed ambiguous
```

hughesjj committed:
Error in aerospike integration test seems unrelated to testcontainers-go upgrade

hughesjj committed:
lintfix

atoulme commented:
This is an occurrence of https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/31001

atoulme commented:
I have kicked of a rerun of the tests.

hughesjj committed:
go tidy

TylerHelmuth commented on receiver/aerospikereceiver/integration_test.go:
```diff
@@ -27,6 +27,7 @@ import (
 var aerospikePort = "3000"
 
 func TestIntegration(t *testing.T) {
+	t.Skip("Container image results in error '/entrypoint.sh: line 35: FEATURE_KEY_FILE: unbound variable'")
```
Is this a result of the upgrade or a previous issue?

TylerHelmuth commented on receiver/hostmetricsreceiver/internal/scraper/networkscraper/network_scraper_test.go:
```diff
@@ -231,5 +231,7 @@ func assertNetworkConnectionsMetricValid(t *testing.T, metric pmetric.Metric) {
 		pcommon.NewValueStr(metadata.AttributeProtocolTcp.String()))
 	internal.AssertSumMetricHasAttribute(t, metric, 0, "state")
-	assert.Equal(t, 12, metric.Sum().DataPoints().Len())
+	// Flaky test gives 12 or 13, so bound it
+	assert.LessOrEqual(t, 12, metric.Sum().DataPoints().Len())
+	assert.GreaterOrEqual(t, 13, metric.Sum().DataPoints().Len())
```
If this is a known flaky test I'd prefer to handle this in a separate PR.

dmitryax replied:
Thanks! merged

atoulme reviewed (approved):
LGTM, agree that the changes for the flaky test should move to a separate PR.

dmitryax commented:
Thanks @hughesjj! We can merge it once the tests skips are resolved or converted into issues with more details to follow-up

hughesjj committed:
try a newer image

hughesjj committed:
update docker in metadataproviders

hughesjj committed:
Should use PostReadies not PostStarts

atoulme reviewed (approved):
LGTM thanks for the fix!

hughesjj committed:
explicitly add goproto api/httpbody and rpc status

hughesjj committed:
make tidy
